File: kerbianbridge/android/KerbianBridge.java
package kerbianbridge.android;

import java.io.*;
import java.net.*;
import org.json.JSONObject;

public class KerbianBridge {
    private ServerSocket serverSocket;
    private Socket clientSocket;
    private BufferedReader in;
    private BufferedWriter out;
    private UIManager uiManager;

    public KerbianBridge(int port) throws IOException {
        serverSocket = new ServerSocket(port);
        clientSocket = serverSocket.accept();
        in = new BufferedReader(new InputStreamReader(clientSocket.getInputStream()));
        out = new BufferedWriter(new OutputStreamWriter(clientSocket.getOutputStream()));
        uiManager = new UIManager();
    }

    public void start() {
        new Thread(() -> {
            String line;
            try {
                while ((line = in.readLine()) != null) {
                    JSONObject msg = new JSONObject(line);
                    handleMessage(msg);
                }
            } catch (Exception e) {
                e.printStackTrace();
            }
        }).start();
    }

    private void handleMessage(JSONObject msg) throws Exception {
        String type = msg.getString("type");
        if (type.equals("call") && msg.getString("target").equals("UIManager")) {
            String method = msg.getString("method");
            JSONObject args = msg.getJSONObject("args");
            if (method.equals("create_view")) {
                int viewId = uiManager.createView(args.getString("type"), args.getJSONObject("props"));
                JSONObject resp = new JSONObject();
                resp.put("type", "response");
                resp.put("id", msg.optInt("id"));
                resp.put("result", new JSONObject().put("view_id", viewId));
                send(resp);
            }
            // ... handle other methods
        }
        // ... handle other targets and message types
    }

    private void send(JSONObject resp) throws IOException {
        out.write(resp.toString());
        out.write("\n");
        out.flush();
    }
}

File: kerbianbridge/ios/KerbianBridge.m
#import "KerbianBridge.h"
#import "UIManager.h"

@implementation KerbianBridge {
    NSFileHandle *_readHandle;
    NSFileHandle *_writeHandle;
    UIManager *_uiManager;
}

- (instancetype)initWithPort:(int)port {
    self = [super init];
    if (self) {
        // Create socket server at port, accept connection
        // Assign _readHandle and _writeHandle
        _uiManager = [[UIManager alloc] init];
    }
    return self;
}

- (void)start {
    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
        while (true) {
            NSData *data = [_readHandle availableData];
            if (data.length == 0) break;
            NSString *line = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
            [self handleMessage:line];
        }
    });
}

- (void)handleMessage:(NSString *)line {
    NSData *jsonData = [line dataUsingEncoding:NSUTF8StringEncoding];
    NSDictionary *msg = [NSJSONSerialization JSONObjectWithData:jsonData options:0 error:nil];
    NSString *type = msg[@"type"];
    if ([type isEqualToString:@"call"] && [msg[@"target"] isEqualToString:@"UIManager"]) {
        NSString *method = msg[@"method"];
        NSDictionary *args = msg[@"args"];
        if ([method isEqualToString:@"create_view"]) {
            NSInteger viewId = [_uiManager createView:args[@"type"] props:args[@"props"]];
            NSDictionary *resp = @{
                @"type": @"response",
                @"id": msg[@"id"],
                @"result": @{@"view_id": @(viewId)}
            };
            NSData *respData = [NSJSONSerialization dataWithJSONObject:resp options:0 error:nil];
            [_writeHandle writeData:respData];
            [_writeHandle writeData:[@"\n" dataUsingEncoding:NSUTF8StringEncoding]];
        }
        // ... handle other methods
    }
    // ... handle other targets and message types
}

@end